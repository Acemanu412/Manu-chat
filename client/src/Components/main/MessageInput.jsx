import React, { useState } from 'react';
import { Form, FormInput, Button } from 'shards-react';
import { useMutation } from '@apollo/client';

import { POST_MESSAGE } from '../../Mutations/messages';

const MessageInput = ({ user, setUser }) => {
    const [message, setMessage] = useState('');
    const [postmessage] = useMutation(POST_MESSAGE)

    const handleSubmit = (e) => {
        e.preventDefault();
        if(!e.target[0].value) return null;
        setMessage('');
        postmessage({ variables: {user: user, content: message }})
    };

    return (
        <div className='input'>
            <Form inline={true} onSubmit={(e) => handleSubmit(e)}>

                    <FormInput 
                    className={'messageForm'}
                    placeholder="Write your message!"
                    label="Message"
                    value={message}
                    onChange={(e) => setMessage(e.target.value)}
                    />

                    <FormInput 
                    className={'userForm'}
                    placeholder="Name"
                    label="User"
                    value={user}
                    onChange={(e) => setUser(e.target.value)}
                    />

                <Button type='submit'>Send</Button>
            </Form>
        </div>
    )
};

export default MessageInput;