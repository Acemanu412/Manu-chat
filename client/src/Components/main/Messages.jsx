import React from 'react';
import { Alert } from 'shards-react';
import { GET_MESSAGES } from '../../Queries/messages';
import {  useSubscription } from '@apollo/client';

const Messages = ({ user }) => {
    const { loading, error, data } = useSubscription(GET_MESSAGES)

    if (loading) return <Alert theme='info'>Loading...</Alert>;
    if (error) return <Alert theme='danger'>Error!</Alert>;

    return (data.messages.map(({id, user: messageUser, content}) => (
      <div key={id} className={`message ${user === messageUser ? 'mine' : 'notMine'}`}> 
        {user !== messageUser && (<div className='userLetters'>{messageUser.slice(0,2).toUpperCase()}</div>)}
        <div className={`messageContent ${user === messageUser ? 'mine' : 'notMine'}`}>{content}</div>
      </div>
    )))
};

export default Messages;