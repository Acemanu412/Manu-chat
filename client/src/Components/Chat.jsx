import React, { useState } from "react";
import 'cross-fetch/polyfill';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
import {Container} from 'shards-react';

import MessagesComponent from './main/Messages';
import MessageInput from './main/MessageInput';

const client = new ApolloClient({
  uri: 'http://localhost:4000/',
  cache: new InMemoryCache()
});

const Chat = () => {
  const [user, setUser] = useState('');

  return <ApolloProvider client={client}>
    <Container className='chatContainer'>
      <h1>CHAT</h1>
      <MessagesComponent user={user} />
      <MessageInput user={user} setUser={setUser} />
    </Container>
  </ApolloProvider>
}
export default Chat;
