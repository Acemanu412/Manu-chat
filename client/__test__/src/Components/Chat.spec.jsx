import React from 'react';
import { shallow } from 'enzyme';
import Chat from '../../../src/Components/Chat';
import Messages from '../../../src/Components/main/Messages';
import MessageInput from '../../../src/Components/main/MessageInput';

describe('Chat component', () => {
  const wrapper = shallow(<Chat />);
  it('should render', () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render Messages Component', () => {
    expect(wrapper.find(Messages).length).toBe(1);
  })

  it('should render MessageInput Component', () => {
    expect(wrapper.find(MessageInput).length).toBe(1);
  })
});