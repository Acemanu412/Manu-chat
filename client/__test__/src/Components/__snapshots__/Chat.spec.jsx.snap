// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component should render 1`] = `"<div class=\\"chatContainer container\\"><h1>CHAT</h1><div class=\\"alert alert-info fade\\" role=\\"alert\\">Loading...</div><div class=\\"input\\"><form class=\\"form-inline\\"><input placeholder=\\"Write your message!\\" label=\\"Message\\" value=\\"\\" class=\\"messageForm form-control\\"/><input placeholder=\\"Name\\" label=\\"User\\" value=\\"\\" class=\\"userForm form-control\\"/><button type=\\"submit\\" class=\\"btn btn-primary\\">Send</button></form></div></div>"`;
